Building On Windows?

Hi, Ive been trying to build the project on Windows and made a lot of headway learning what would be necessary to build mlir-hs with minimal pain; I was curious if anyone had any insights or suggestions to limit any issues (like the fact building LLVM dylibs are currently unsupported on Windows.)

I got further with building on Windows with a simplified approach. The manual steps I described is what I would now call "the hard way". You can leverage mingw directly to simplify installation (Works for any version before version 18, you have to build LLVM manually in MinGW to get later versions at time of writing): stack exec -- pacman -S mingw-w64-x86_64-mlir. However this leads to a different error, in which it can't find an llvm.lib and I'm currently unsure if any form of .dll could be linked. I intend to continue experimenting.

This is the error:
```
PS C:\Users\Me\OneDrive\Desktop\Projects\mlir-hs-main> stack build
mlir-hs> configure (lib)
C:/Users/Me/AppData/Local/Programs/stack/x86_64-windows/ghc-8.10.4/mingw/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lLLVM
collect2.exe: error: ld returned 1 exit status
```

--- a/Setup.hs
+++ b/Setup.hs
@@ -16,6 +16,7 @@ import Data.Char
 import Data.List
 import Data.Maybe
 
+import System.Info
 import System.Directory
 import System.FilePath
 
@@ -84,9 +85,10 @@ buildTblgen confFlags = do
   cppFlags   <- words <$> llvmConfig ["--cppflags"]
   includeDir <- trim  <$> llvmConfig ["--includedir"]
   cc <- getCC confFlags
+  let windowsLLVMVersion = "-" ++ (show $ head $ versionNumbers llvmVersion)
   ensureDirectory $ cwd </> ".bin"
   cc $ sources ++ cxxFlags ++ ldFlags ++
-        [ "-lMLIR", "-lLLVM", "-lMLIRTableGen", "-lLLVMTableGen"
+        [ "-lMLIR", if os == "mingw32" then "-lLLVM" ++ windowsLLVMVersion else "-lLLVM", "-lMLIRTableGen", "-lLLVMTableGen"
         , "-o", cwd </> ".bin/mlir-hs-tblgen"]
   let tblgenProgram = ConfiguredProgram
         { programId           = "mlir-hs-tblgen"

